fl4health/datasets/skin_cancer/load_data.py

```diff
@@ -51,49 +71,79 @@ def load_skin_cancer_data(
+        train_transform (Union[None, Callable]): Transformations to apply to the training data. Defaults to None.
+        val_transform (Union[None, Callable]): Transformations to apply to the validation data. Defaults to None.
+        test_transform (Union[None, Callable]): Transformations to apply to the test data. Defaults to None.
+        dataset_converter (Optional[DatasetConverter]): Converter to apply to the dataset. Defaults to None.
+        seed (Optional[int]): Random seed for shuffling data. Defaults to None.
+
+    Returns:
+        Tuple[DataLoader, DataLoader, DataLoader, Dict[str, int]]: DataLoaders for the training, validation,
+            and test sets, and a dictionary with the number of examples in each set.
+    """
+    if sum(split_percents) != 1.0:
+        raise ValueError("The split percentages must sum to 1.0")
 
     dataset_paths = {
-        "Barcelona": data_dir / "ISIC_2019" / "ISIC_19_Barcelona.json",
-        "Rosendahl": data_dir / "HAM10000" / "HAM_rosendahl.json",
-        "Vienna": data_dir / "HAM10000" / "HAM_vienna.json",
-        "UFES": data_dir / "PAD-UFES-20" / "PAD_UFES_20.json",
-        "Canada": data_dir / "Derm7pt" / "Derm7pt.json",
+        "Barcelona": data_dir.joinpath("ISIC_2019", "ISIC_19_Barcelona.json"),
+        "Rosendahl": data_dir.joinpath("HAM10000", "HAM_rosendahl.json"),
+        "Vienna": data_dir.joinpath("HAM10000", "HAM_vienna.json"),
+        "UFES": data_dir.joinpath("PAD-UFES-20", "PAD_UFES_20.json"),
+        "Canada": data_dir.joinpath("Derm7pt", "Derm7pt.json"),
     }
 
     if dataset_name not in dataset_paths:
-        log(ERROR, f"Dataset {dataset_name} not found in available datasets.")
         raise ValueError(f"Dataset {dataset_name} not found in available datasets.")
 
     dataset_path = dataset_paths[dataset_name]
 
     if not dataset_path.exists():
-        log(
-            ERROR,
+        raise FileNotFoundError(
             f"Dataset file {dataset_path} does not exist.\
-            Please follow the instructions in fl4health/datasets/skin_cancer/README.md.",
+            Please follow the instructions in fl4health/datasets/skin_cancer/README.md."
         )
-        raise FileNotFoundError(f"Dataset file {dataset_path} does not exist.")
 
-    log(INFO, f"Data directory: {str(dataset_path)}")
+    print(f"Data directory: {str(dataset_path)}")
 
     with open(dataset_path, "r") as f:
         data = json.load(f)["data"]
 
-    random.seed(seed)
+    if seed is not None:
+        random.seed(seed)
     random.shuffle(data)
+
     total_size = len(data)
-    train_size = int(0.7 * total_size)
-    valid_size = int(0.15 * total_size)
+    train_size = int(split_percents[0] * total_size)
+    valid_size = int(split_percents[1] * total_size)
 
     train_data = data[:train_size]
     valid_data = data[train_size : train_size + valid_size]
+    test_data = data[train_size + valid_size :]
+
+    val_test_transform = transforms.Compose(
```

Comment:
Maybe just add a comment that this is the default transform if more specific ones aren't defined?

Reply:
added!